dvc/repo/metrics/show.py

```diff
@@ -136,7 +136,7 @@ def _read_metrics(self, metrics, branch):
         if not metric:
             continue
 
-        res[out.rel_path] = metric
+        res[out.rel_path] = (metric, typ)
```

Comment:
@efiop , is there a better way to achieve this?

Reply:
I could use `https://docs.python.org/2/library/csv.html#csv.Sniffer` to detect if the file is separated by comas, semicolons, or tabs.

So I don't need the type at all :thinking:

Reply:
Need to test for "raw" type with commas and "json" with comas (maybe adding some more cases).

Dang, maybe I should've created different commits instead of amending :thinking: